### ISSUE TITLE
Toolbar Buttons Are Not Operable by Keyboard on Plone5b4

### ISSUE THREAD

**Issue by ma10** [plone/Products.CMFPlone#866]
Plone5 beta 4 seems to have issues with keyboard operation on some of the toolbar buttons.

I assume if you hit Enter on the "Add New" or "State: published," a submenu would show up, but no changes that can be detected by screen reader happens.

How to reproduce the problem:
1. Install the beta4 using the unified installer.
2. create a new Plone site.
3. Access the top page of the newly created site.
4. Hit the Enter key on these buttons (technically links, I suppose.)

Tested environments:
- Firefox 40.03 (32bit) with JAWS 14.0 on Windows
- Firefox 40.03 (32bit) with NVDA 2015.3 on Windows
- Google Chrome 45.0.2454.85 m (64-bit) with JAWS 14.0 on Windows
- Google Chrome 45.0.2454.85 m (64-bit) with NVDA 2015.3 on Windows

**Comment by ma10** [comment]
ps: This is also the case with the 'Translate' button when you enable plone.app.multilingual and add another language as a translatable language.

**Comment by polyester** [comment]
@ma10 I am now checking on Plone5  rc2, using NVDA 2015.3, and the submenus do open correctly. Apart from annoying habit of having bulletpoints in front of them in the submenu it works as I would expect.

Cannot check in Jaws as I don't own it.

**Comment by ma10** [comment]
@polyester:

I see, it does work if I do the following with NVDA (though not with JAWS):

1. Hit enter on one of the buttons, e.g. "Add New."
2. (NVDA switches to the "focus mode"), and hit the Tab key to move to the submenu items.

However, if you switch back to the browse mode by hitting Ins+SPC before hitting the tab key, it seems like NVDA can't recognize the submenu items.  Users often expect that they can find the submenu items using the up/down cursor keys in the browse mode.

If it can be changed so that you can find the submenu items in NVDA's browse mode, I think the issue goes away with JAWS as well.

**Comment by polyester** [comment]
Thanks, I still need to learn a lot about how to operate NVDA properly (and one day JAWS). Will investigate.

**Comment by ma10** [comment]
I played around a little more and found out the following.

* JAWS's Forms Mode, or the virtual PC cursor off (which is equivalent to NVDA's focus mode) acts the same as NVDA's focus mode.
* Regardless if you hit enter on toolbar buttons, if you keep hitting the tab key, you'll reach, and go through the submenu items.

This is in a sense better than how it used to be as with NVDA's browse mode or JAWS's virtual PC cursor, you don't have to read all the submenu items in order to go through the toolbar buttons.

But if you are not used to Plone's toolbar buttons and submenus, or you happen to know that yo would need to use the browse mode, you are quite lost.

Since I'm not an expert on the WAI-ARIA, I'm not too certain what's the best way to go about this, but I believe each button with submenu items should have "menu" for the role attribute and "true" for the aria-haspopup attribute, while submenu items should have "menuitem" for their role attributes.  And aria-hidden attribute on submenu items must be adjusted accordingly.

Foe example, the simple dropdown menu example on [this page](http://heydonworks.com/practical_aria_examples/) seems to work fine.
